src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Abstractions/IGeneratedSymbolConfig.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Collections.Generic;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Abstractions
+{
+    /// <summary>
+    /// Represents the configuration of <see cref="IGeneratedSymbolMacro"/> and <see cref="IGeneratedSymbolMacro{T}"/>.
+    /// </summary>
+    public interface IGeneratedSymbolConfig : IMacroConfig
```

Comment:
I was thinking about that, but wanted to discuss entrypoint first.
What would be indication that deterministic mode should be used? should we
- bypass it through `IGenerator`?
- use env variable? 
- anything else

Reply:
Eventually it should probably get passed from env, but ideally only as high component as possible relies on the env (so it probably shouldn't be Generator nor TemplateCreator - the info about determinist mode should be passed to them).
It might as well be passed via `IEngineEnvironmentSettings` (as we have env snapshot there)

Reply:
Implemented handling of environment variables in `MacroProcessor`.